### PR TITLE
packages/awscli-v2: Multiple updates and new versions for dependent package

### PR DESCRIPTION
**Pull request by teaguesterling**
This addresses a few things that came up while adding a new version of the `awscli-v2` package:
1. Adding a new version resulted in issues with the `py-ruamel-yaml-clib`, which is optional in python>=3.9
2. The `post_install` block failed due to (I suspect) changes to the spack API. Reordering the decorators fixed it.
3. The `py-awscrt` package was a strict version requirement, which was different than the previous version.
4. The `py-awscrt` package itself needed the required version. I also took the opportunity to add some new ones.
5. Added a `list_url` to make future updates easier.

### CODE REVIEW

File: var/spack/repos/builtin/packages/awscli-v2/package.py
Comment (climbfuji) [inline var/spack/repos/builtin/packages/awscli-v2/package.py]: Does this change make a difference? If not, why should it be changed?

Comment (teaguesterling) [reply var/spack/repos/builtin/packages/awscli-v2/package.py]: The post_install step was consistently failing for me with a spack error, thus causing the install to fail. Making this change resolved it.

Comment (alalazo) [reply var/spack/repos/builtin/packages/awscli-v2/package.py]: Can you change the two lines above with:
```suggestion
    @run_after("install", when="~examples")
```
? The when argument of `run_after` was added exactly to avoid remembering the ordering.

Also, note that right now the `when` is not taken into consideration - `run_after` registers a function for later execution, and since the function is not `when` decorated the `when` part doesn't count.

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/awscli-v2/package.py
Comment (teaguesterling) [reply var/spack/repos/builtin/packages/awscli-v2/package.py]: Thanks for providing this suggestion and clarifying the logic!

**Review by alalazo — APPROVED** [review]
Waiting for @climbfuji review before merging

**Review by climbfuji — APPROVED** [review]
Thanks for these updates and for volunteering as a maintainer!

**Pull request merged**
